include/nbl/video/IGPUCommandPool.h

```diff
@@ -31,11 +35,21 @@ class NBL_API IGPUCommandPool : public core::IReferenceCounted, public IBackendO
         // Vulkan: const VkCommandPool*
         virtual const void* getNativeHandle() const = 0;
 
+        virtual void reset()
+        {
+            assert(!getCreationFlags().hasFlags(ECF_RESET_COMMAND_BUFFER_BIT));
```

Comment:
You can have `ECF_RESET_COMMAND_BUFFER_BIT` and reset command pool at the same time.
unless it's an internal reference holding reason I'm not aware of.
We should be able to both reset command buffers indvidually(if `ECF_RESET_COMMAND_BUFFER_BIT`) or reset the whole command pool.